openwisp_radius/views.py

```diff
@@ -0,0 +1,23 @@
+from private_storage.views import PrivateStorageDetailView
+from swapper import load_model
+
+from .base.models import _get_csv_file_private_storage
+
+
+class RadiusBatchCsvDownloadView(PrivateStorageDetailView):
+    storage = _get_csv_file_private_storage
+    model = load_model('openwisp_radius', 'RadiusBatch')
+    model_file_field = 'csvfile'
+    slug_field = 'csvfile'
+    slug_url_kwarg = 'csvfile'
+
+    def can_access_file(self, private_file):
+        user = private_file.request.user
+        if not user.is_authenticated:
+            return False
+        org = (self.object.organization.pk,)
+        user_organizations = user.organizations_pk.all()
+        return user.is_superuser or (user.is_staff and org in user_organizations)
```

Comment:
does this take in consideration if the user is flagged as `is_admin` of the organization?

I noticed that also this doesn't do it: https://github.com/openwisp/openwisp-firmware-upgrader/blob/master/openwisp_firmware_upgrader/private_storage/views.py and opened an issue for it https://github.com/openwisp/openwisp-firmware-upgrader/issues/90

Reply:
Thanks, the different testcases make much more sense now! :smile: